Finish type signatures of get_optional_..._attributes

ajnelson-nist opened the pull request:
This patch series also goes through type safety checks from some of the freshly annotated data flows.

Last, one typo is fixed.

ajnelson-nist committed:
Finish type signatures of get_optional_..._attributes

This patch also goes through type safety checks from some of the freshly
annotated data flows.

Last, one typo is fixed.

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist commented:
@fabrizio-turchi , FYI - The issues found this time by CI and fixed in this branch were also found by running `make check` locally.  The Makefile is written to behave very closely to CI.

fabrizio-turchi commented:
Thanks. The only weird thing is that the code:

```python
from .lib import JSONLD, get_attribute, get_optional_integer_attribute, ...
```
locally raises the error:

```python
ImportError: attempted relative import with no known parent package
```

ajnelson-nist commented:
Ah, dear.  Apologies, I forgot to smoke-test the GUI.

ajnelson-nist commented:
Reopening, reverting to draft while adding another test and fix.

ajnelson-nist committed:
Add --dry-run to GUI

This patch updates CI to run up to before initiating the GUI as a smoke
test.

This patch identifies an issue reported by Fabrizio.  By itself, the
patch will fail CI.  A fix is coming in a follow-on patch.

Reported-by: Fabrizio Turchi <fabrizio.turchi@igsg.cnr.it>
Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist committed:
Update poetry call in CI

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist committed:
Update poetry call in CI

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist commented:
Error case reproduced in [`f753f32`](https://github.com/casework/CASE-Application-Navigator/pull/12/commits/f753f329f5510b3619bc78cdb688c696626d9211).

ajnelson-nist commented:
@fabrizio-turchi , the issue being reported is one we saw previously on the `CASE-Mappings-Python` repository.  Relative imports are a surprisingly complex matter between the testing infrastructure and the ability to run the code with this specific command-line form:

```bash
python3 case_viewer/case_viewer.py .../input.jsonld
```

_If_ the code can be run as an installed command, then the testing and user access can be harmonized.  I.e., if this becomes an OK call pattern ...

```bash
case_viewer .../input.jsonld
```

... then the testing can remain.  Unfortunately, for several technical reasons (which I decline to discuss on GitHub), the current call pattern of `python3 case_viewer/case_viewer.py` is not compatible with the test infrastructure I had added.

I think it would be appropriate to switch the prescribed call style to `case_viewer .../input.jsonld`, because of the dependencies on QT that already necessitate having an augmented Python environment.  (That is, I expect a virtual environment will typically be necessary for this application.)

However, I will defer to others' desired call style on this matter.  Fabrizio, which would you prefer be the manner in which this application is called?

1. `python3 case_viewer/case_viewer.py .../input.jsonld`
2. `case_viewer .../input.jsonld`

fabrizio-turchi commented:
The form case_viewer .../input.jsonld is fine with me. 
Please merge the type_review_get_optional__attributes branch into the main one.

ajnelson-nist commented:
Thank you.  I'll need to add another patch reflecting the now-acknowledged capability.

ajnelson-nist committed:
Change invocation pattern from Python script call to CLI command

This patch carries out type revisions entailed by defining a `main()`
method.  Some of the global variables that were `if`-`__main__`-gated
needed to be moved outside of the `if` gate.  It seems as further
functions serving as subroutines get type-reviewed, `global` statements
will be needed.

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist committed:
Port Make test updates to CI

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist commented:
CI passes.  Merging with Fabrizio's emailed permission.
